### DISCUSSION TITLE
VFD on S905X & S905W

### DISCUSSION THREAD

**Discussion by fachos** [devmfc/debian-on-amlogic#32] [Ideas]
Hi dev, 

Can you add tm1628 available to generic s905x/s905w DTB?

thank you

**Comment by devmfc** [comment]
No, than they are no longer generic, kind of defeats the purpose 😀. 

Not all tvboxes have VFD displays, and if they do, they are not all TM1628 and many of them use different GPIO's. 

The same is true for LEDs and buttons.

Try one of the specific dtb's and see if that works, or modify the dtb yourself. If you know what dtb settings are necessary to make it work, I can make a dtb and config for it for a next release.

**Comment by fachos** [comment]
Hi... yes not all tvbox have the display... i extract the one of the dtb copy & modify s905x generic dtb back then try 
but still no VFD show up... dmesg also show errors...

tvbox:~#cat meson-sc2-x96x4-gbit.dts

        tm1628 {
                compatible = "devmfc,tm1628";
                dat-gpios = <0x09 0x00 0x06>;
                clk-gpios = <0x09 0x01 0x06>;
                stb-gpios = <0x09 0x27 0x06>;
                status = "okay";

tvbox:~#dmesg | grep tm1628

[    0.109616] OF: /tm1628: could not get #gpio-cells for /soc/interrupt-controller@c4301000
[    0.109645] OF: /tm1628: could not get #gpio-cells for /soc/interrupt-controller@c4301000
[    0.109672] OF: /tm1628: could not get #gpio-cells for /soc/interrupt-controller@c4301000
[    0.227942] tm16xx tm1628: DEVMFC: tm16xx driver v1.96 probe
[    0.227998] OF: /tm1628: could not get #gpio-cells for /soc/interrupt-controller@c4301000
[    0.228028] tm16xx tm1628: Get clk GPIO failed: -22
[    0.228041] tm16xx: probe of tm1628 failed with error -22


i have the config for my tvbox display

--------------------------------------------------------------------------------
vfd_gpio_clk='1,4,0'
vfd_gpio_dat='1,5,0'
vfd_gpio_stb='1,3,0'
vfd_chars='4,0,1,2,3'
vfd_dot_bits='0,1,2,3,4,5,6'
--------------------------------------------------------------------------------

i dont know how to convert my config to your code
maybe you can help modify to make it work...  

thank you

**Comment by devmfc** [comment]
You will need to have the android DTB's to make a vfd config for your boxes. What box do you have?

**Comment by fachos** [comment]
Hi.. i have 2 tvbox

1- tanix tx3 mini (s905w 2GB ram 16GB storage)
2- x95pro (s905x 2GB ram 16GB storage)

Can you modify and add my config to your DTB?
i run armbian before with openvfd installed. 
everytime armbian load openvfd will read the config file..
if have different box.. just need change/modify the config file

![image](https://github.com/devmfc/debian-on-amlogic/assets/12787402/37fb59ed-e471-413a-8a8e-c62a86f4cd12)
and
![image](https://github.com/devmfc/debian-on-amlogic/assets/12787402/f9607706-8cdb-4585-b5f5-a4731b1c7c80)

**Comment by fachos** [comment]
Hi dev...

 🙏

**Reply by LITUATUI** [reply]
Hi. Did you managed to get the VFD working?
